corehq/apps/users/tasks.py

```diff
@@ -338,10 +338,17 @@ def process_reporting_metadata_staging():
                     num_unsent_forms=record.num_unsent_forms,
                     num_quarantined_forms=record.num_quarantined_forms
                 )
+                if not record.last_heartbeat:
+                    # coming from sync
+                    latest_build_date = record.sync_date
+                else:
+                    # coming from hearbeat
+                    latest_build_date = record.modified_on
```

Comment:
This now [reflects the code](https://github.com/dimagi/commcare-hq/pull/26305/files#diff-291dfd3802bf0daa51e598788d5ca13bL343) as it was before I modified it in this PR